### PR TITLE
[ARTS-293] Create Person Validation automation api tests

### PR DESCRIPTION
**Pull request by sameerapurini**
The PR covers the api automation tests written on mocha chai api framework written using javascript. Mochawesome reports are generated from the automation test run.

The ticket, https://ndph-arts.atlassian.net/browse/ARTS-293 is a test approach that lists out the wide variety of tests / scenarios considered to check the practitioner service end point responses.

**Note:** Dependant on arts-184 per merge

### CODE REVIEW

File: api-tests/config/conf.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+global.expect = require('chai').expect
+global.request = require('supertest')
+global.HttpStatus = require('http-status-codes')
+
+module.exports = {
+    BASE_URL: process.env.BASE_URL,
+    reporter: 'node_modules/mochawesome',
+    'reporter-option': [
+        'overwrite=true',
+        'reportTitle=My\ Custom\ Title',
```
Comment (balteravishay) [inline api-tests/config/conf.js:10]: Report title

File: api-tests/specs/practitioner-services/createperson.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const { validPerson, invalidCharacterLength, missingGivenName, missingfamilyName, missingPrefix } = require('../../data/createPerson')
+const conf = require('../../config/conf')
+
+beforeEach(function () {
+    baseRequest = request(conf.BASE_URL)
+})
+
+describe('As a user with Create Person permission, I want to have my create person request validated by the system, So that I cannot create an invalid person', function () {
```
Comment (balteravishay) [inline api-tests/specs/practitioner-services/createperson.js:8]: Some more cases we can test for:
- Illegal characters (see the regex)
- Unresponsive (not existing) fhir store 
- Malformed input

**Comment by balteravishay** [comment]
In addition to the comments in the code itself, a general comment.
Since much of how the service works depends on configuration, we should plan to think of a way to inject different congratulations to the system under test, as a way to increase coverage of its functionality.

File: api-tests/config/conf.js (hunk 1 above)
Comment (sameerapurini) [reply api-tests/config/conf.js:10]: I will cover that section once the api tests are successfully run through the pipeline 👍

File: api-tests/specs/practitioner-services/createperson.js (hunk 2 above)
Comment (sameerapurini) [reply api-tests/specs/practitioner-services/createperson.js:8]: Hi @balteravishay I have added the illegal characters check scenario, not sure how to cover the unresponsive (FHIR) store, please advise. Also I have already covered malformed input as part of other scenarios.

Comment (sameerapurini) [reply api-tests/specs/practitioner-services/createperson.js:8]: With regards to the 'Malformed input' the invalid json format is not acceptable and replicating an unresponsive FHIR store would just show a 404 which I am not sure would add any value to the tests.

**Review by mdp0 — APPROVED** [review]

File: api-tests/data/createperson.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const validPerson = {
```
Comment (tamirkamara) [inline api-tests/data/createperson.js:1]: is it also possible to have the data files arranged to folders just as you do in the specs?

File: api-tests/specs/practitioner-services/createperson.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+const { validPerson, invalidCharacterLength, missingGivenName, missingfamilyName, missingPrefix } = require('../../data/createPerson')
```
Comment (tamirkamara) [inline api-tests/specs/practitioner-services/createperson.js:1]: I think the folder name should be "practitioner-service"

**Comment by balteravishay** [comment]
Looks like there are some merge issues

File: api-tests/specs/practitioner-services/createperson.js (hunk 2 above)
Comment (sameerapurini) [reply api-tests/specs/practitioner-services/createperson.js]: Hi @balteravishay the malformed script isn't showing a 400 when run using mocha but somehow postman is able to show it. I added the script which can anyways assert it.

**Comment by sameerapurini** [comment]
Closing the PR as merge conflicts are unresolvable.

**Pull request closed without merging**